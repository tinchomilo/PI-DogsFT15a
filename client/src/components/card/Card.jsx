import React from 'react'
import imgDefault from '../../resources/imgFoto.jpg';
import { Link } from 'react-router-dom'

import style from '../../styles/card.module.css'


export const Card = ( { name, img, temperament, temperaments, id } ) => {
    return (
        <div className={ style.card }>
            <h3>{name}</h3> 
            <Link to={ `/detail/${ id }`}>
            <img 
                src={ img ? img : imgDefault } 
                alt='imagen raza' width="150" height="150"/>
            <div className={ style.names }>                                              
                    <p>Temperamento: { temperament ? temperament.map( elem => elem + ', ') :
                            temperaments?.map( elem => elem.name + ', ') 
                        }
                    </p>                
            </div>
            </Link>
        </div>
    )
}
