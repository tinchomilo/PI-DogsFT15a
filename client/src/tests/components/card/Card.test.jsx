import React from 'react'
import { shallow } from 'enzyme';

import { Card } from '../../../components/card/Card';

describe('Pruebas en el componente card', () => {

  const name = 'Terry'
  const img = 'https://localhost/imagenTerry.jpg'
  const temperament = ['curious', 'playful']
  const id = '508'

  // inicio los valores afuera del test ya que no los voy a cambiar
  const wrapper = shallow( <Card 
                              name= {name} 
                              img= { img } 
                              temperament= { temperament } 
                              id={ id } /> );

    test('debe mostrar el componente correctamente', () => {

      expect( wrapper).toMatchSnapshot();

    })

    test('debe mostrar un h3 con el nombre', () => {

      const nameH3 = wrapper.find( 'h3' ).text()
      expect( nameH3 ).toBe( name )

    })

    test('debe de mostrar la imagen corectamente', () => {
      
      const imagen = wrapper.find( 'img' )
      expect( imagen.prop( 'src' ) ).toBe( img )

    })

    test('debe de tener la clase card', () => {

      const div = wrapper.find( 'div' )
      // console.log(div.at(0).props()) como hay m√°s de 1 div devuelve un arreglo y por eso tengo que usar el at(0) ya que es el div que quiero evaluar
      // console.log(div.at(0).prop('className'))
      const className = div.at(0).prop( 'className' )

      expect( className.includes( 'card' ) ).toBe( true )

    })
    
  
})
