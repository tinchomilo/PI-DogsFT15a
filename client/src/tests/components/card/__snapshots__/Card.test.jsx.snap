// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente card debe mostrar el componente correctamente 1`] = `
<div
  className="card"
>
  <h3>
    Terry
  </h3>
  <Link
    to="/detail/508"
  >
    <img
      alt="imagen raza"
      height="150"
      src="https://localhost/imagenTerry.jpg"
      width="150"
    />
    <div
      className="names"
    >
      <p>
        Temperamento: 
        curious, 
        playful, 
      </p>
    </div>
  </Link>
</div>
`;
